pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -146,118 +146,245 @@ func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStart
 
 }
 
-// shouldEnsureSuggested checks if the exempt priority level exists and returns
-// whether the suggested flow schemas and priority levels should be ensured.
-func shouldEnsureSuggested(flowcontrolClientSet flowcontrolclient.FlowcontrolV1beta1Interface) (bool, error) {
-	if _, err := flowcontrolClientSet.PriorityLevelConfigurations().Get(context.TODO(), flowcontrol.PriorityLevelConfigurationNameExempt, metav1.GetOptions{}); err != nil {
-		if apierrors.IsNotFound(err) {
-			return true, nil
-		}
-		return false, err
+const thisFieldManager = "api-priority-and-fairness-config-producer-v1"
+
+// fsAutoUpdateStrategyFunc accepts the current and desired FlowSchema configuration and determines whether an update is necessary.
+// current is the existing configuration stored.
+// desired is the desired spec we want stored.
+// On return, shouldUpdate is set to true if auto update is required and newObject will represent the new configuration to be stored.
+type fsAutoUpdateStrategyFunc func(current, desired *flowcontrolv1beta1.FlowSchema) (newObject *flowcontrolv1beta1.FlowSchema, shouldUpdate bool)
```

Comment:
Why not simply have one return value, `nil` for no update?

Reply:
it seems more idiomatic to use `*flowcontrolv1beta1.FlowSchema, bool`